ci_scripts/publish_target_validation_report.py

```diff
@@ -0,0 +1,35 @@
+import logging
+import tempfile
+import dotenv
+import os
+from target_validation_report import PlatformValidator
+from aws_helpers import upload_file
+
+logger = logging.getLogger(__name__)
+
+
+def generate_validation_report_and_publish():
+    with tempfile.TemporaryDirectory() as tmp_directory:
+        platform_validator = PlatformValidator(tmp_directory, True)
+        platform_validator.render_results()
+        if platform_validator.processing_error:
+            e = Exception("Report generation failed")
+            logger.error(e)
+            raise e
+        upload_file(os.path.join(tmp_directory, 'index.html'), 'validation')

```

Comment:
I'm not sure if the `PlatformValidator` interface is here to blame, but having implicit `index.html` as side effect is a bit smelly - if the output of `PlatformValidator` changes, this might end up being a hard to spot bug. I wonder if it wouldn't be more future proof to upload the entire contents of `tmp_directory` instead (i.e. not target `index.html` explicitly). 🤔

Reply:
good point
Will do